mm.lic

```diff
@@ -0,0 +1,94 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#mm
+=end
+custom_require.call(%w[common-arcana events])
+
+class MoonMagery
+  include DRCA
+  def initialize
+    args = get_args
+    @settings = get_settings
+
+    if args.locate
+      locate?(args.person)
+    elsif args.rift
+      rift(args.person)
+    end
+    Flags.delete('locate-fail')
+  end
+
+  def get_args
+    arg_defs = [
+      [
+        { name: 'locate', regex: /locate/i },
+        { name: 'person', regex: /\w+/, description: 'Name of person to locate' }
+      ],
+      [
+        { name: 'rift', regex: /rift/i },
+        { name: 'person', regex: /\w+/, description: 'Name of person to rift' }
+      ]
+    ]
+
+    parse_args(arg_defs)
+  end
+
+  def rift(person)
+    person = person.capitalize
+    check_moonwatch
+    unless UserVars.moons.any? { |moon_name, moon_data| UserVars.moons['visible'].include?(moon_name) && moon_data['timer'] >= 0 }
+      echo 'No moons available to cast Rift'
+      return
+    end
+    data = if @settings.waggle_sets['rs']
```

Comment:
I'm not sure how I feel about using abbreviations to this degree in a waggle set name... what do you think?

Reply:
Im following a naming convention for single spell sets where the name of the set is the spell abbreviation. This is a personal preference for convenience and lets all spells be cast via `;buff <abbrev>` for convenience. I am happy to use some other convention.